core/src/test/java/org/elasticsearch/VersionTests.java

```diff
@@ -262,18 +270,30 @@ public void testLuceneVersionIsSameOnMinorRelease() {
             }
         }
     }
-    private  static final Version V_20_0_0_UNRELEASED = new Version(20000099, Version.CURRENT.luceneVersion);
+
+    private static final Version V_20_0_0_UNRELEASED = new Version(20000099, Version.CURRENT.luceneVersion);
 
     // see comment in Version.java about this test
     public void testUnknownVersions() {
         assertUnknownVersion(V_20_0_0_UNRELEASED);
-        // once we release 5.0.2 and it's added to Version.java we need to remove this constant
+        // once we release 5.0.2 and it's added to Version.java, we need to remove this constant
         assertUnknownVersion(ElasticsearchException.V_5_0_2_UNRELEASED);
+        // once we release 5.1.0 and it's added to Version.java, we need to remove these constants
+        assertUnknownVersion(ElasticsearchException.V_5_1_0_UNRELEASED);
```

Comment:
I was wondering too, if this is the way we do it at this point. It feels weird that we have 5.1.0 constant till now, and all of a sudden we have to change all classes that were using it. I understand why we do this in master, not sure about 5.x. What's the reasoning behind this?